pandas/tests/indexes/common.py

```diff
@@ -205,6 +205,39 @@ def test_hash_error(self):
                                        type(ind).__name__):
                 hash(ind)
 
+    def test_copy(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, MultiIndex):
+                continue
+
+            first = index.__class__(index, copy=True, name='mario')
+            second = first.__class__(first, copy=False)
+
+            self.assertTrue(index.equals(first))
+            # Even though "copy=False", we want a new object:
+            self.assertIsNot(first, second)
+
+            self.assertEqual(first.name, 'mario')
+            self.assertEqual(second.name, 'mario')
+
+    def test_ensure_copied_data(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, (PeriodIndex, RangeIndex, MultiIndex)):
+                # Cannot be instantiated from just _data as first argument
+                continue
+
+            index_type = index.__class__
+            idx1 = index_type(index.values, copy=True)
+            self.assert_numpy_array_equal(index.values, idx1.values)
```

Comment:
instead of the ``assert_numpy_array_equal`` do

``tm.assert_index_equal(index, idx1)``

also rename ``idx1`` -> ``result``; more consistent with other tests nomenclature